src/formatDate.js

```diff
@@ -50,7 +50,53 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const toSeperator = toFormat[toFormat.length - 1];
+  const fromSeperator = fromFormat.pop();
+  const dateArray = date.split(fromSeperator);
+  let dateYear, dateMonth, dateDay;
+  const dateOutput = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      dateYear = dateArray[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      dateMonth = dateArray[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      dateDay = dateArray[i];
+    }
+  }
+
+  if (dateYear.length === 2) {
+    if (parseInt(dateYear) < 30) {
+      dateYear = `20${dateYear}`;
+    } else {
+      dateYear = `19${dateYear}`;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'YYYY') {
+      dateOutput[dateOutput.length] = dateYear;
+    }
+
+    if (toFormat[i] === 'YY') {
+      dateOutput[dateOutput.length] = dateYear.slice(2);
+    }
+
+    if (toFormat[i] === 'MM') {
+      dateOutput[dateOutput.length] = dateMonth;
+    }
+
+    if (toFormat[i] === 'DD') {
+      dateOutput[dateOutput.length] = dateDay;
+    }
```

Comment:
Great place for switch case. You check only one parameter and you have limited number of outcomes.

Reply:
Not fixed.